devito/archinfo.py

```diff
@@ -187,24 +187,30 @@ def simd_items_per_reg(self, dtype):
 
 class Cpu64(Platform):
 
-    def _detect_isa(self):
-        return 'cpp'
-
-
-class Intel64(Cpu64):
+    # The known isas ​​will be overwritten in the specialized classes
+    known_isas = tuple()
 
     def _detect_isa(self):
-        known_isas = ['cpp', 'sse', 'avx', 'avx2', 'avx512']
-        for i in reversed(known_isas):
+        for i in reversed(self.known_isas):
             if any(j.startswith(i) for j in get_cpu_info()['flags']):
                 # Using `startswith`, rather than `==`, as a flag such as 'avx512'
                 # appears as 'avx512f, avx512cd, ...'
                 return i
         return 'cpp'
 
 
+class Intel64(Cpu64):
+
+    def _detect_isa(self):
+        self.known_isas = ('cpp', 'sse', 'avx', 'avx2', 'avx512')
```

Comment:
no, this doesn't make any sense conceptually

I don't know the reason you didn't pass continuous integration, I looked at it quickly but definitely it was unrelated to your changes. Can you rollback to the previous commit and push again?

you should also learn and use `git rebase` at this point,  because now we have 6 commits for what is supposed to be one simple change. [Here](https://git-scm.com/docs/git-rebase) is a good starting point

we can talk about it maybe even later

Reply:
I will investigate the continuous investigation failure (@mloubout : actually, it was really weird stuff in `test_symbol_cache`......)

Reply:
rollback done